src/y-webrtc.js

```diff
@@ -466,96 +466,130 @@ const openRoom = (doc, provider, name, key) => {
 const publishSignalingMessage = (conn, room, data) => {
   if (room.key) {
     cryptoutils.encryptJson(data, room.key).then(data => {
-      conn.send({ type: 'publish', topic: room.name, data: buffer.toBase64(data) })
+      conn.publish(room.name, buffer.toBase64(data))
     })
   } else {
-    conn.send({ type: 'publish', topic: room.name, data })
+    conn.publish(room.name, data)
   }
 }
 
-export class SignalingConn extends ws.WebsocketClient {
+export class SignalingConn {
   constructor (url) {
-    super(url)
+    this.url = url
     /**
      * @type {Set<WebrtcProvider>}
      */
     this.providers = new Set()
-    this.on('connect', () => {
-      log(`connected (${url})`)
-      const topics = Array.from(rooms.keys())
-      this.send({ type: 'subscribe', topics })
-      rooms.forEach(room =>
-        publishSignalingMessage(this, room, { type: 'announce', from: room.peerId })
-      )
+    this.setupClient()
+  }
+
+  setupClient() {
+    this.client = new ws.WebsocketClient(this.url)
+    this.client.on('connect', () => {
+      log(`connected (${this.url})`)
+      this.handleConnect()
     })
-    this.on('message', m => {
+    this.client.on('message', m => {
       switch (m.type) {
         case 'publish': {
-          const roomName = m.topic
-          const room = rooms.get(roomName)
-          if (room == null || typeof roomName !== 'string') {
-            return
+          this.handleMessage(m.topic, m.data)
+        }
+      }
+    })
+    this.client.on('disconnect', () => log(`disconnect (${this.url})`))
+  }
+
+  handleConnect() {
+    const topics = Array.from(rooms.keys())
+    topics.forEach(topic =>
+      this.subscribe(topic)
+    )
```

Comment:
Azure Web PubSub cannot join multiple groups, but that shouldn't stop the built-in client from doing so. Maybe instead of the `forEach` this needs a `subscribeMulti()` method or let `subscribe()` detect whether `topic` is an array in derived classes and do the iteration there?